openhands/server/session/agent_session.py

```diff
@@ -160,6 +170,38 @@ def _create_security_analyzer(self, security_analyzer: str | None):
                 security_analyzer, SecurityAnalyzer
             )(self.event_stream)
 
+    def _clone_repo(self, github_token: str | None, selected_repository: str | None):
+        if not github_token or not selected_repository or not self.runtime:
+            return
+        url = f'https://{github_token}@github.com/{selected_repository}.git'
+        dir_name = selected_repository.split('/')[1]
+        action = CmdRunAction(
+            command=f'git clone {url} {dir_name} ; cd {dir_name} ; git checkout -b openhands-workspace'
+        )
+        logger.info(action, extra={'msg_type': 'ACTION'})
+        self.runtime.run_action(action)
+
+    def _add_custom_microagents(self, agent: Agent, selected_repository: str | None):
```

Comment:
If we end up putting this on the runtime class, I'd suggest refactoring this a bit to return `custom_microagents_content` so that the runtime doesn't need to get access to the Agent.